quantinuum-hugr/src/hugr/validate.rs

```diff
@@ -419,16 +421,18 @@ impl<'a, 'b> ValidationContext<'a, 'b> {
 
         let is_static = match from_optype.port_kind(from_offset).unwrap() {
             EdgeKind::Static(typ) => {
-                if !(OpTag::Const.is_superset(from_optype.tag())
-                    || OpTag::Function.is_superset(from_optype.tag()))
-                {
-                    return Err(InterGraphEdgeError::InvalidConstSrc {
+                let reqd_optype = match typ {
```

Comment:
TBH I think this check is more of an assertion about the spec, than something we really need to check at runtime; there is no other way to construct nodes with such out-ports, and as such I don't think it's even possible to test the error path here. So I'd be happy to remove both these checks and the relevant error variant....?